adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/DatasourceManager.java

```diff
@@ -64,30 +53,31 @@ public Datasource createDatasource(Datasource config) {
 
 
   @Transactional
-  public void updateDatasource(Long id, Datasource updated) throws IllegalArgumentException {
-    Datasource old = datasourceRepository.findById(id)
-      .orElseThrow(() -> new IllegalArgumentException("Datasource with id " + id + " not found."));
+  public void updateDatasource(Long id, Datasource update) throws IllegalArgumentException {
+    Datasource existing = datasourceRepository.findById(id)
+      .orElseThrow(() -> new IllegalArgumentException("Datasource with id " + id + " not found"));
 
-    datasourceRepository.save(applyUpdate(old, updated));
-    eventRepository.save(new DatasourceEvent(EventType.DATASOURCE_UPDATE, id));
+    datasourceRepository.save(applyUpdate(existing, update));
+    amqpPublisher.publishUpdate(existing);
   }
 
 
   @Transactional
   public void deleteDatasource(Long id) {
+    Datasource datasource = datasourceRepository.findById(id)
+            .orElseThrow(() -> new IllegalArgumentException("Datasource with id " + id + " not found"));
     datasourceRepository.deleteById(id);
-    eventRepository.save(new DatasourceEvent(EventType.DATASOURCE_DELETE, id));
+    amqpPublisher.publishDeletion(datasource);
   }
 
 
   @Transactional
   public void deleteAllDatasources() {
     Iterable<Datasource> allDatasourceConfigs = getAllDatasources();
     datasourceRepository.deleteAll();
-
-    allDatasourceConfigs.forEach(
-      pl -> eventRepository.save(new DatasourceEvent(EventType.DATASOURCE_DELETE, pl.getId()))
-    );
+    for (Datasource ds: allDatasourceConfigs) {
```

Comment:
use a stream instead, otherwise one unsuccessful deletion publish might block the other ones